File: frontend/types/domain/leaderboard.ts
export interface LeaderboardRow {
  player_id: string;
  first_name: string;
  last_name: string;
  position: number;
  total_score: number;
  relative_to_par: number;
  holes_completed: number;
  is_dnf: boolean;
}

export interface Leaderboard {
  game_id: string;
  status: string;
  rows: LeaderboardRow[];
}

File: backend/app/schemas/leaderboard.py
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict

class LeaderboardRow(BaseModel):
    player_id: UUID
    first_name: str
    last_name: str
    position: int
    total_score: int
    relative_to_par: int
    holes_completed: int
    is_dnf: bool = False

class LeaderboardResponse(BaseModel):
    game_id: UUID
    status: str
    rows: List[LeaderboardRow]
    
    model_config = ConfigDict(from_attributes=True)
